ui/tests/integration/components/tools/wrap-test.js

```diff
@@ -104,6 +109,63 @@ module('Integration | Component | tools/wrap', function (hooks) {
     await click(TS.submit);
   });
 
+  test('it toggles between views and preserves input data', async function (assert) {
+    assert.expect(4);
+    await this.renderComponent();
+    await codemirror().setValue(this.wrapData);
+    assert.dom('[data-test-component="json-editor-title"]').hasText('Data to wrap (json-formatted)');
+    await click('[data-test-toggle-input="json"]');
+    assert.dom('[data-test-component="json-editor-title"]').doesNotExist();
+    await click('[data-test-toggle-input="json"]');
+    assert.dom('[data-test-component="json-editor-title"]').exists();
+    assert.strictEqual(
+      codemirror().getValue(' '),
+      `{   \"foo": \"bar" }`, // eslint-disable-line no-useless-escape
+      'json editor has original data'
+    );
+  });
+
+  test('it submits from kv view', async function (assert) {
+    assert.expect(6);
+
+    const flashSpy = sinon.spy(this.owner.lookup('service:flash-messages'), 'success');
+    const updatedWrapData = JSON.stringify({
+      ...JSON.parse(this.wrapData),
+      foo2: 'bar2',
+      foo3: 'foo\nbar\nbaz',
+    });
+
+    this.server.post('sys/wrapping/wrap', (schema, { requestBody, requestHeaders }) => {
+      const payload = JSON.parse(requestBody);
+      assert.propEqual(payload, JSON.parse(updatedWrapData), `payload contains data: ${requestBody}`);
+      assert.strictEqual(requestHeaders['X-Vault-Wrap-TTL'], '30m', 'request header has default wrap ttl');
+      return {
+        wrap_info: {
+          token: this.token,
+          accessor: '5yjKx6Om9NmBx1mjiN1aIrnm',
+          ttl: 1800,
+          creation_time: '2024-06-07T12:02:22.096254-07:00',
+          creation_path: 'sys/wrapping/wrap',
+        },
+      };
+    });
+
+    await this.renderComponent();
+    await codemirror().setValue(this.wrapData);
+    await click('[data-test-toggle-input="json"]');
+    await fillIn('[data-test-kv-key="1"]', 'foo2');
+    await fillIn('[data-test-kv-value="1"]', 'bar2');
+    await click('[data-test-kv-add-row="1"]');
+    await fillIn('[data-test-kv-key="2"]', 'foo3');
+    await fillIn('[data-test-kv-value="2"]', 'foo\nbar\nbaz');
```

Comment:
this is technically still a single line string, we could save a variable in this test using backticks, and pass that in here instead, i.e:
```js
const topSecretData = `this is my multi-line secret
that I want to wrap
blah blah, something
some seriously important config`